src/components/review-form.js

```diff
@@ -0,0 +1,110 @@
+'use client'
+import { useState } from 'react'
+import styles from './review-form.module.css'
+import { StarRating } from './star-rating'
+import { validateReviewFormValues } from '../lib/validation'
+
+export function ReviewForm({ movieId }) {
+  const [formValues, setFormValues] = useState({
+    name: '',
+    email: '',
+    rating: null,
+    review: '',
+  })
+  const [errors, setErrors] = useState({})
+
+  const [submitStatus, setSubmitStatus] = useState('not_submitted')
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+
+    const errors = validateReviewFormValues(formValues)
+    setErrors(errors)
+    const hasErrors = Object.keys(errors).length > 0
+    if (!hasErrors) {
```

Comment:
The constant `hasErrors` is used only once, but I think it makes the code easier to read.